master/pkg/tasks/task_spec.go

```diff
@@ -238,7 +239,9 @@ func (g GCCheckpoints) Entrypoint() []string {
 
 // Environment implements InnerSpec.
 func (g GCCheckpoints) Environment(TaskSpec) model.Environment {
-	return g.ExperimentConfig.Environment
+	env := model.DefaultExperimentConfig(&g.TaskContainerDefaults).Environment
+	env.EnvironmentVariables = g.ExperimentConfig.Environment.EnvironmentVariables
```

Comment:
Copied the environment vars of the experiment in the off chance someone is relying on `GOOGLE_APPLICATION_CREDENTIALS` being in there or something.